src/app/core/store/shopping/products/products.effects.ts

```diff
@@ -169,18 +172,44 @@ export class ProductsEffects {
     )
   );
 
-  loadProductBundles$ = createEffect(() =>
+  loadProductParts$ = createEffect(() =>
     this.actions$.pipe(
-      ofType(loadProductSuccess),
-      mapToPayloadProperty('product'),
-      filter(product => ProductHelper.isProductBundle(product)),
-      mergeMap(({ sku }) =>
-        this.productsService.getProductBundles(sku).pipe(
-          mergeMap(({ stubs, bundledProducts }) => [
-            ...stubs.map((product: Product) => loadProductSuccess({ product })),
-            loadProductBundlesSuccess({ sku, bundledProducts }),
-          ]),
-          mapErrorToAction(loadProductFail, { sku })
+      ofType(loadProductParts),
+      mapToPayloadProperty('sku'),
+      groupBy(identity),
+      mergeMap(group$ =>
+        group$.pipe(
+          this.throttleOnBrowser(),
+          mergeMap(sku =>
+            this.store.pipe(
+              select(getProductParts(sku)),
+              first(),
```

Comment:
Why is first() and take(1) used in the same pipe? Doesn't the take(1) make first() irrelevant?

Reply:
Not in this case. Every store select must be aborted after the first result, otherwise the effect keeps on firing when the result is set. In this case though, the following filter would also implicitly prevent that.

Reply:
Good to have it explicit and it's nice to have this comment for future reference in case someone wonders :D